content/ide-integration/_index.md

```diff
@@ -3,15 +3,120 @@ title: "IDE Integration"
 weight: 8
 ---
 
+
+## Downloading the source
+
+Either git clone the tinygo source from.
+
+> https://github.com/tinygo-org/tinygo
+
+## Go mod init in your project
+
+Navigate to your project and type:
+
+> go mod init
+
+## Replace needed imports
+
+Replace each std package you need using the replace keyword in the go.mod file.
+
+So if you want the machine package to be resolved replace it using.
+
+> replace machine => /path/to/tinygo/machine
+
+Repeat this step for every package, which needs to be resolved in your project.
+
+### Example
+
+```go
+module github.com/Nerzal/tinygo-playground
+
+go 1.14
+
+replace machine => /home/tobias/go/src/github.com/tinygo-org/tinygo/src/machine
+
+require (
+    machine v0.0.0-00010101000000-000000000000
+    tinygo.org/x/drivers v0.13.0
+)
+
+```
+
+## Add go module files to the packages
+
+In order to get your package resolved, you also have to add a go mod file for the packages.
+
+So navigate to `/path/to/tinygo/src`
+And then do `go mod init`
+
+### Example
+
+```bash
+ cd ~/go/src/github.com/tinygo-org/tinygo/src/machine
+ go mod init
+```
+
 ## GoPath
 
 In order to get IDE support like autocomplete you have to add the tinygo src path to your [GOPATH](https://github.com/golang/go/wiki/GOPATH).
 After adding the tinygo source path to the GOPATH go will know, where to look for the packages like `machine`
 
 ### Ubuntu Example
+
 You can just append the path to your tinygo installation in your GOPATH.
 
->  export GOPATH=$GOPATH:/path/to/your/tinygo
+> export GOPATH=$GOPATH:/path/to/your/tinygo
```

Comment:
If you don't have GOPATH set, you may need to check with go-env and set the GOPATH first.

Reply:
For example, the following must be configured.

```
export GOPATH=`go env GOPATH`:/path/to/your/tinygo
```

Reply:
So is your suggestion to use that command instead of the given command?

Reply:
It's difficult because I think everyone has different settings.
If $GOPATH is already set, you can just use the original description.


However, I think some people don't have $GOPATH set, so I would suggest adding an annotation.

Reply:
> > echo $GOPATH
> If you result is empty, you need to set the GOPATH fist

You've already added.
Thank you.